crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -732,9 +726,50 @@ pub(super) fn from_radix(s: &Series, radix: u32, strict: bool) -> PolarsResult<S
     let ca = s.utf8()?;
     ca.parse_int(radix, strict).map(|ok| ok.into_series())
 }
-pub(super) fn str_slice(s: &Series, start: i64, length: Option<u64>) -> PolarsResult<Series> {
-    let ca = s.utf8()?;
-    Ok(ca.str_slice(start, length).into_series())
+
+pub(super) fn str_slice(s: &[Series]) -> PolarsResult<Series> {
+    let ca = s[0].utf8()?;
+
+    let s1 = &s[1];
+    let s2 = &s[2];
+
+    polars_ensure!(
+        s1.len() <= ca.len(),
+        ComputeError:
+        "too many `offset` values ({}) for column length ({})",
+        s1.len(), ca.len(),
+    );
+
+    polars_ensure!(
+        s2.len() <= ca.len(),
+        ComputeError:
+        "too many `length` values ({}) for column length ({})",
+        s2.len(), ca.len(),
+    );
+
+    let offset = match s1.len() {
+        1 => {
+            let offset = s1.get(0).unwrap();
+            s1.clear().extend_constant(offset, ca.len()).unwrap()
+        },
+        _ => s1.clone(),
+    };
+
+    let offset = offset.cast(&DataType::Int64)?;
+    let offset = offset.i64()?;
+
+    let length = match s2.len() {
```

Comment:
I don't want to add broadcasting by extending memory. See discussion here; #11900

Reply:
Ah! Excellent, thanks @ritchie46 - that explains the need for the added length checks / fallback logic. 

I'll switch it around.